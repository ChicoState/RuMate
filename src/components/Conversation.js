import React, { useState } from 'react';
import { View, Text, StyleSheet, TouchableOpacity, Alert } from 'react-native';
import * as Haptics from 'expo-haptics';

const Conversation = ({ name, blurb, navigation }) => {
  const [render, setRender] = useState(true)
  if (render) {
    return (
      <TouchableOpacity 
        onPress={() => {
          Haptics.selectionAsync();
          navigation.navigate('Conversation', {name})
        }}
        onLongPress={() => {
          Haptics.impactAsync()
          Alert.alert(
            'Delete messages?',
            'Tap delete to remove the conversation',
            [
              { text: 'Delete', onPress: () => {
                setRender(false)
              }},
              { text: 'Cancel', onPress: () => val = false, style:'cancel'}
            ]
          )
        }}
      >
        <View style={styles.container}>
          <Text style={styles.name}>{name}</Text>
          <Text style={styles.blurb}>{blurb}</Text>
        </View>
      </TouchableOpacity>
    );
  } else {
    return null
  }
}

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 25,
    paddingBottom: 10,
    borderColor: '#111',
    borderBottomWidth: 1,
    paddingVertical: 20
  },
  name: {
    fontSize: 18,
    paddingVertical: 5,
  },
  blurb: {
    fontSize: 10,
    paddingLeft: 15,
  }
});

export default Conversation;