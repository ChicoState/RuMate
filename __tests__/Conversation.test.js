import React from 'react';
import renderer from 'react-test-renderer';

import Conversation from '../src/components/Conversation';

describe('<App />', () => {

  it('has 1 child', () => {
    const tree = renderer.create(<Conversation />).toJSON();
    console.log(tree.children.length)
    expect(tree.children.length).toBe(1);
  });
  
  it('renders correctly', () => {
    const tree = renderer.create(<Conversation />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
});

