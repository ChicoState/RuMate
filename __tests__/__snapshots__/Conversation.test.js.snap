// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#111",
        "paddingBottom": 10,
        "paddingHorizontal": 25,
        "paddingVertical": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
          "paddingVertical": 5,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 10,
          "paddingLeft": 15,
        }
      }
    />
  </View>
</View>
`;
